frappe/core/doctype/user_type/user_type.py

```diff
@@ -36,16 +36,19 @@ def set_modules(self):
 		if not self.user_doctypes:
 			return
 
-		modules = frappe.get_all("DocType",
-			fields=["module"],
-			filters={"name": ("in", [d.document_type for d in self.user_doctypes])},
-			distinct=True,
-		)
+		DocType = frappe.qb.DocType("DocType")
+
+		document_types = [d.document_type for d in self.user_doctypes] or ['']
+		modules = (frappe.qb.from_(DocType)
```

Comment:
I think we need to refactor this in the frappe.get_all() method. So that all frappe.get* parameters continue to work as before.

Reply:
@TurkerTunali  Can you specify your concerns with this query conversion?

Reply:
Here we are getting modules. I am not sure but instead of refoctoring here by eliminating the "frappe.get_all()" and using "frappe.qb.DocType()" we should refactor this in "frappe.get_all()" method only. 

Most probably we are using frappe.get_all() everywhere and it will require significant effort to refactor everywhere.